modules/imap/hm-imap.php

```diff
@@ -425,9 +425,13 @@ public function get_mailbox_list($lsub=false, $mailbox='', $keyword='*') {
                 $marked = false;
                 $folder_sort_by = 'ARRIVAL';
                 $check_for_new = false;
-
+                
                 /* full folder name, includes an absolute path of parent folders */
-                $folder = $this->utf7_decode($vals[(count($vals) - 1)]);
+                if ($vals[(count($vals) - 1)] == ')' && $lsub) {
+                    $folder = $this->utf7_decode($vals[6]);
```

Comment:
Why is this hardcoded value of 6?

Reply:
when lsub is true and a folder has sub folders, $vals[(count($vals) - 1)] does not contain the folder name as it contains more keys. That why I used index 6 because the folder name is always at that key when it comes to lsub.

<img width="277" alt="Screenshot 2023-07-02 at 08 24 26" src="https://github.com/cypht-org/cypht/assets/80334370/3d01e706-783c-43b0-9297-db9934781b05">

Reply:
It might be always index 6 with your tests but I doubt this is the case for every imap server as I don't see it in the specs - you should use the RFC when it comes to IMAP protocol: https://datatracker.ietf.org/doc/html/rfc3501#section-6.3.9
Maybe the LSUB result contains some more flags at the end but still, there should be a standard way to get the folder name out of the list based on strict reference documentation and not on arbitrary array indexing.

Reply:
Can you paste here the full contents of $vals in your case?